libs/image/include/image/ImageSampler.h

```diff
@@ -159,6 +161,20 @@ uint32_t getMipmapCount(const LinearImage& source);
 UTILS_PUBLIC
 Filter filterFromString(const char* name);
 
+/**
+ * Returns whether simpleScaleDownRgba() is usable on the specified dimensions.
+ */
+bool canSimpleScaleDown(uint32_t sourceWidth, uint32_t sourceHeight, uint32_t destinationWidth, uint32_t destinationHeight);
+
+/**
+ * Scales down a single byte per component RGBA image where each destination pixel corresponds to
+ * a whole number of source pixels on each dimension. canSimpleScaleDown(sourceWidth, sourceHeight, destinationWidth, destinationHeight)
+ * must return true as a precondition.
+ */
+std::unique_ptr<uint8_t[]> simpleScaleDownRgba(
+    const uint8_t *source, uint32_t sourceWidth, uint32_t sourceHeight, uint32_t sourceLineStride,
+    uint32_t destinationWidth, uint32_t destinationHeight, uint32_t destinationLineStride);
+
```

Comment:
don't put this in public headers, this is only internal machinery.

Reply:
moved these to the `StbProvider`, not exposed publicly anymore